listenbrainz/webserver/views/profile.py

```diff
@@ -202,3 +212,33 @@ def upload():
 
         flash.info('Congratulations! Your listens from %d files have been uploaded successfully.' % success)
     return redirect(url_for("profile.import_data"))
+
+
+@profile_bp.route('/request-stats', methods=['GET'])
+@login_required
+def request_stats():
+    """ Check if the current user's statistics have been calculated and if not,
+        put them in the stats queue for stats_calculator.
+    """
+    status = _redis.redis.get(construct_stats_queue_key(current_user.musicbrainz_id)) == 'queued'
+    if status == 'queued':
+        flash.info('You have already been added to the stats calculation queue! Please check back later.')
+    elif db_stats.valid_stats_exist(current_user.id):
+        flash.info('Your stats were calculated in the most recent stats calculation interval,'
+            ' please wait until the next interval! We calculate new statistics every Monday at 00:00 UTC.')
+    else:
+        # publish to rabbitmq queue that the stats-calculator consumes
+        data = {
+            'type': 'user',
+            'id': current_user.id,
+            'musicbrainz_id': current_user.musicbrainz_id,
+        }
+        publish_data_to_queue(
+            data=data,
+            exchange=current_app.config['STATS_EXCHANGE'],
+            queue=current_app.config['STATS_QUEUE'],
+            error_msg='Could not put user %s into statistics calculation queue, please try again later',
+        )
+        _redis.redis.set(construct_stats_queue_key(current_user.musicbrainz_id), 'queued')
```

Comment:
Now that we merged the brainzutils PR, should this be updated to use brainz util caching instead?

Reply:
This isn't exactly caching which is why I used our RedisConnection directly similar to how we use it for storing now_playing listens.